File: python/07-26/20-07-26 Golden Ratio.py
"""
Golden Ratio
Given two numbers, determine if their ratio approximates the golden ratio.

- Use a golden ratio of 1.618
- Allow a tolerance of 0.01

1. is_golden_ratio(21, 34) should return True.
2. is_golden_ratio(15, 20) should return False.
3. is_golden_ratio(8, 13) should return True.
4. is_golden_ratio(10, 16) should return False.
5. is_golden_ratio(1618, 1000) should return True.
6. is_golden_ratio(88, 55) should return False.
"""

def is_golden_ratio(a: int, b: int) -> bool:
    golden_ratio: float = 1.618
    tolerance: float = 0.01
    ratio_ab: float = max(a, b) / min(a, b)

    return abs(golden_ratio - ratio_ab) <= tolerance


print(is_golden_ratio(21, 34))
print(is_golden_ratio(15, 20))
print(is_golden_ratio(8, 13))
print(is_golden_ratio(10, 16))
print(is_golden_ratio(1618, 1000))
print(is_golden_ratio(88, 55))

File: javascript/07-26/20-07-26 Golden Ratio.js
/*
Golden Ratio
Given two numbers, determine if their ratio approximates the golden ratio.

- Use a golden ratio of 1.618
- Allow a tolerance of 0.01

1. is_golden_ratio(21, 34) should return True.
2. is_golden_ratio(15, 20) should return False.
3. is_golden_ratio(8, 13) should return True.
4. is_golden_ratio(10, 16) should return False.
5. is_golden_ratio(1618, 1000) should return True.
6. is_golden_ratio(88, 55) should return False.
 */

function isGoldenRatio(a, b) {
    const goldenRatio = 1.618;
    const tolerance = 0.01;
    const ratio = Math.max(a, b) / Math.min(a, b);

    return Math.abs(goldenRatio - ratio) <= tolerance;
}


console.log(isGoldenRatio(21, 34));
console.log(isGoldenRatio(15, 20));
console.log(isGoldenRatio(8, 13));
console.log(isGoldenRatio(10, 16));
console.log(isGoldenRatio(1618, 1000));
console.log(isGoldenRatio(88, 55));
